src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -816,10 +832,11 @@ private void CompileSynthesizedSealedAccessors(SourcePropertySymbol sourceProper
             if ((object)synthesizedAccessor != null && !_globalHasErrors)
             {
                 Debug.Assert(synthesizedAccessor.SynthesizesLoweredBoundBody);
-                var discardedDiagnostics = DiagnosticBag.GetInstance();
+                var discardedDiagnostics = BindingDiagnosticBag.GetInstance();
                 synthesizedAccessor.GenerateMethodBody(compilationState, discardedDiagnostics);
                 Debug.Assert(!discardedDiagnostics.HasAnyErrors());
-                discardedDiagnostics.Free();
+                discardedDiagnostics.DiagnosticBag.Clear();
+                _diagnostics.AddRangeAndFree(discardedDiagnostics);
```

Comment:
Consider using a method that is explicit about adding Dependencies only. Calling clear on the line before this is confusing at first read.

Reply:
> Consider using a method that is explicit about adding Dependencies only. Calling clear on the line before this is confusing at first read.

This is addressed in [https://github.com/dotnet/roslyn/pull/40640](https://github.com/dotnet/roslyn/pull/40640)

---
In reply to: [362136679](https://github.com/dotnet/roslyn/pull/40560#discussion_r362136679) [](ancestors = 362136679)